.github/workflows/compatibility-updater.yaml

```diff
@@ -0,0 +1,61 @@
+name: Utility / Compatibility Matrix Update
+
+on:
+  schedule:
+    # Runs at 00:00 UTC every day
+    - cron: '0 0 * * *'
+
+jobs:
+  run-compatibility-updater:
+    runs-on: ubuntu-latest
+
+    env:
+      SCRIPTS_DIR: utils/compatibility
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        # Install pyenv dependencies
+        sudo apt-get update
+        sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+            libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+            libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev \
+            liblzma-dev
+
+    - name: Install pyenv
```

Comment:
could we not use something like: https://github.com/marketplace/actions/pyenv-build

Reply:
We could, but it doesn't necessarily improve the runtime speed or flow.
I've scripted the Python env setup to be dynamic and repeatable locally and in the GitHub action. 
Doing it via the script simplifies the maintenance of local dev and won't require any changes to the workflow yaml

Reply:
oh no it just makes it easier to set up pyenv in the github action i would expect, instead of manually doing it ourselves.

Reply:
Would you prefer we use that action?
I can make the changes and test using it.

Reply:
I made the changes to use the Pyenv Builder Action and reopened the PR

Reply:
It's using an older version of pyenv, so I had to downgrade the python version from `3.12.3` to `3.12.0`, everything tested fine though